stable/velero/templates/secret.yaml

```diff
@@ -0,0 +1,16 @@
+{{- if and .Values.credentials.useSecret (not .Values.credentials.existingSecret) -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "velero.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "velero.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ include "velero.chart" . }}
+type: Opaque
+data:
+{{- range $key, $value := .Values.credentials.secretContents }}
```

Comment:
I had trouble getting this format to work, though I may have been doing something wrong...

Here's what worked for me, and the command I used to do the install:

```
{{- if and .Values.credentials.useSecret (not .Values.credentials.existingSecret) -}}
apiVersion: v1
kind: Secret
metadata:
  name: {{ include "velero.fullname" . }}
  labels:
    app.kubernetes.io/name: {{ include "velero.name" . }}
    app.kubernetes.io/instance: {{ .Release.Name }}
    app.kubernetes.io/managed-by: {{ .Release.Service }}
    helm.sh/chart: {{ include "velero.chart" . }}
type: Opaque
data:
  cloud: |-
    {{ .Values.credentials.secretContents | b64enc }}
{{- end }}
```

`helm install --name velero -f values.yaml --set-file credentials.secretContents=/home/nrb/credentials-velero-gke ./velero`

Reply:
My only concern here is that `cloud: |` is now hard-coded in. Is there any case where a secret would not start with that? I took the secrets template exactly from the old Ark chart, so it should work?

I've used the secret template as it is in the PR now (`{{- range $key, $value := .Values.credentials.secretContents }}`) and didn't have any issues with it. My secret in values.yaml looked something like this
```
  secretContents:
    cloud: |
      [default]
      aws_access_key_id = ABC123
      aws_secret_access_key = ABC123
```

Reply:
That's true that it would be hard coded...all of our documentation includes it, but I don't think it's mandatory.


I was unable to get my GCP JSON working in my own values.yaml with the current code (and hadn't in the old Ark template), but I believe it may have been an indenting issue. I'll look more today as it's likely better to keep the current `range` functionality, as you say, and I'm formatting my values incorrectly.